datastore/longhorn.go

```diff
@@ -1249,6 +1283,57 @@ func (s *DataStore) ListNodesRO() ([]*longhorn.Node, error) {
 	return s.nLister.Nodes(s.namespace).List(labels.Everything())
 }
 
+func (s *DataStore) ListNodesHaveEngineImage(image string) (map[string]*longhorn.Node, error) {
+	nodes, err := s.ListNodes()
+	if err != nil {
+		return nil, err
+	}
+	eiName := types.GetEngineImageChecksumName(image)
+	ei, err := s.GetEngineImage(eiName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get engine image %v: %v", image, err)
+	}
+	if ei.Status.State != types.EngineImageStateDeployed && ei.Status.State != types.EngineImageStateDeploying {
+		return nil, fmt.Errorf("engine image %v (%v) is in state %v", ei.Name, image, ei.Status.State)
```

Comment:
Do we need to error or can we just return an empty list?

Reply:
Yeah, return an empty map is better. 
Thank you. Fixed